test/common/services/security/system_indices_user.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { systemIndicesSuperuser } from '@kbn/test';
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+const SYSTEM_INDICES_SUPERUSER_ROLE = 'system_indices_superuser';
+
+export async function createSystemIndicesUser(ctx: FtrProviderContext) {
+  const log = ctx.getService('log');
+  const config = ctx.getService('config');
+  const es = ctx.getService('esWithoutSystemIndices');
+
+  const enabled = !config
+    .get('esTestCluster.serverArgs')
+    .some((arg: string) => arg === 'xpack.security.enabled=false');
+
+  if (enabled) {
+    log.debug('===============creating system indices role and user===============');
+
+    await es.security.putRole({
+      name: SYSTEM_INDICES_SUPERUSER_ROLE,
+      cluster: ['all'],
+      indices: [
+        {
+          names: ['*'],
+          privileges: ['all'],
+          allow_restricted_indices: true,
+        },
+      ],
+      applications: [
+        {
+          application: '*',
+          privileges: ['*'],
+          resources: ['*'],
+        },
+      ],
+      run_as: ['*'],
+    });
+
+    await es.security.putUser({
+      username: systemIndicesSuperuser.username,
+      password: systemIndicesSuperuser.password,
+      roles: [SYSTEM_INDICES_SUPERUSER_ROLE],
+    });
+  }
```

Comment:
__question:__ I assume this change will [break this test](https://github.com/elastic/kibana/blob/main/x-pack/test/functional/apps/security/users.ts#L40-L61) in Cloud since we expect only `test_user` at the moment (spotted via https://github.com/elastic/kibana/issues/63276 last time). If so, would you mind updating that test as well? I think we should stop relying on test users in this test and do something like this instead:


```ts
// In Cloud default users are defined in file realm, such users aren't exposed through the Users API.
if (isCloudEnvironment()) {
	expect('elastic' in users).to.be(false);
	expect('kibana_system' in users).to.be(false);
	expect('kibana' in users).to.be(false);
} else {
	expect(users.elastic.roles).to.eql(['superuser']);
	expect(users.elastic.reserved).to.be(true);
	expect(users.elastic.deprecated).to.be(false);

	expect(users.kibana_system.roles).to.eql(['kibana_system']);
	expect(users.kibana_system.reserved).to.be(true);
	expect(users.kibana_system.deprecated).to.be(false);

	expect(users.kibana.roles).to.eql(['kibana_system']);
	expect(users.kibana.reserved).to.be(true);
	expect(users.kibana.deprecated).to.be(true);
}
```

Reply:
Good spot @azasypkin! I've checked this test on my cloud cluster and it indeed failed as you assumed.
I've applied your suggested change in d0f5ac5 with a small modification to get a more verbose failure message (`Error: expected true to equal false` is not ideal to debug).

Reply:
Great, thank you @pheyos!